secure/storage/vault/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::collections::HashMap;
+use thiserror::Error;
+use ureq;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum Error {
+    #[error("Http error: {1}")]
+    HttpError(u16, String),
+    #[error("Internal error: {0}")]
+    InternalError(String),
+    #[error("Missing field {0}")]
+    MissingField(String),
+    #[error("404: Not Found: {0}/{1}")]
+    NotFound(String, String),
+    #[error("Serialization error: {0}")]
+    SerializationError(String),
+}
+
+impl From<std::io::Error> for Error {
+    fn from(error: std::io::Error) -> Self {
+        Self::SerializationError(format!("{}", error))
+    }
+}
+
+impl From<ureq::Response> for Error {
+    fn from(response: ureq::Response) -> Self {
+        Error::HttpError(response.status(), response.status_line().into())
+    }
+}
+
+impl From<serde_json::error::Error> for Error {
+    fn from(error: serde_json::error::Error) -> Self {
+        Self::SerializationError(format!("{}", error))
+    }
+}
+
+/// Client provides a client around the restful interface to a Vault servce. Learn more
+/// here: https://www.vaultproject.io/api-docs/
+pub struct Client {
+    host: String,
+    token: String,
```

Comment:
Doesn't need to be done now, but FYI tokens have an expiry and need to be periodically renewed by the client.

Reply:
Yes! I realized that too when reading from token API. Considering the complexity here, we should see how hard it is to make an indefinite versus a rotation framework for tokens. Sigh.